docs/Processes.md

```diff
@@ -555,6 +555,349 @@ The modifications from the standard tick.q include:
+
+### Time Zone Behaviour
+
+A key tickerplant function is to timestamp the incoming data before it gets published to the rest of the system. Similar to the existing TP, the STP allows definition of 
+
+- the timezone that data is timestamped at upon arrival
+- the timezone that the STP executes it's end-of-day roll in 
+- the offset from midnight that the STP executes its end-of-day roll in
+
+This allows for more complex configurations, such as an Foreign Exchange data capture system which timestamps data in UTC but rolls at 5 PM EST. 
+
+The key variable used is `.eodtime.dailyadj` but more information on setting up a TorQ process in a different time zone can be found [here](https://aquaqanalytics.github.io/TorQ/utilities/#eodtimeq).
+
+### Per Table Customisation
+
+Each table has its own upd function, meaning that some additional processing, such as adding a sequence number or a time-zone offset, can be done in the STP itself rather than needing to be done in a separate process. This is done by altering the `.stplg.updtab` dictionary in the segmentedtickerplant settings config file. The default behaviour is for every update to automatically have the current timestamp applied to it.
+
+```q
+// In file $TORQHOME/appconfig/settings/segmentedtickerplant.q
+// Apply a sequence number to 'tabname'
+.stplg.updtab[`tabname]:{((count first x)#'(y;.stplg.seqnum),x}
+
+// In the STP process
+q) .stplg.updtab
+quote   | {(enlist(count first x)#y),x}
+trade   | {(enlist(count first x)#y),x}
+tabname | {((count first x)#'(y;.stplg.seqnum),x}
+...
+```
+
+### Custom Batching Modes
+
+The batching behaviour depends on two functions: `.u.upd` and `.z.ts`. The former is called every time an update arrives in the STP and the latter whenever the timer function is called (e.g. if the process is started with `-t 1000`, this will be called every second). In the default batching mode, `.u.upd` inserts the update into a local table and writes it to the log file, and `.z.ts` publishes the contents of the local table before clearing it. To customise these functions, the `.stplg.upd` and `.stplg.zts` dictionaries will need to be customised. For example, the default batching code looks like the following:
+
+```q
+\d .stplg
+
+// Standard batch mode - write to disk immediately, publish in batches
+upd[`defaultbatch]:{[t;x;now]
+  t insert x:.stplg.updtab[t] . (x;now);
+  `..loghandles[t] enlist(`upd;t;x);
+  // track tmp counts, and add these after publish
+  @[`.stplg.tmpmsgcount;t;+;1];
+  @[`.stplg.tmprowcount;t;+;count first x];
+ };
+
+zts[`defaultbatch]:{
+  // publish and clear all tables, increment counts
+  .stpps.pubclear[.stpps.t];
+  // after data has been published, updated the counts
+  .stplg.msgcount+:.stplg.tmpmsgcount;
+  .stplg.rowcount+:.stplg.tmprowcount;
+  // reset temp counts
+  .stplg.tmpmsgcount:.stplg.tmprowcount:()!();
+ };
+
+\d .
+```
+
```

Comment:
```suggestion
### Multiple Updates From Publishers
The STP handles single messages which contain multiple updates. The standard interface for publishers is the `.u.upd` interface supported by the original tickerplant (two arguments, the table to be updated followed by the data). The arguments for this can now but lists, i.e. a list of table names followed by a list of associated data. These updates can all be stamped with the same sequence number, and is useful where upstream publishers receive a single update which is split across downstream tables.
```